src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -121,19 +133,34 @@ def __init__(  # pylint: disable=too-many-arguments
             )
 
     @override
-    async def _run(self) -> None:  # pylint: disable=too-many-locals
-        """Run actor main function.
+    async def _run(self) -> None:
+        """Run this actor's logic.
+
+        It waits for new power requests and process them. Only one power request
+        can be processed at a time to prevent from sending multiple requests for
+        the same components to the microgrid API at the same time.
 
-        It waits for new requests in task_queue and process it, and send
-        `set_power` request with distributed power.
-        The output of the `set_power` method is processed.
-        Every battery and inverter that failed or didn't respond in time will be marked
+        A new power request will be ignored if a power request with the same
+        components is currently being processed.
+
+        Every component that failed or didn't respond in time will be marked
         as broken for some time.
         """
         await self._component_manager.start()
 
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            req_id = frozenset(request.component_ids)
+
+            if req_id in self._processing_tasks:
+                if pending_request := self._pending_requests.get(req_id):
+                    _logger.warning(
```

Comment:
A small one:

```suggestion
                    _logger.debug(
```

Otherwise I think it could get pretty spammy. There are many reasons why a request could get ignored, so I don't think we need to emit a warning for this, it is normal.